valerymo/integreatly-operator
# Initial commit to replace BBTs with Prometheus probes.

# Issue link
https://issues.redhat.com/browse/MGDAPI-1878

# What
Added logic to create Prometheus Probes instead of Blackbox Targets which are then consumed by Prometheus. As of now, this only applies for single-tenant RHOAM installations.

# Verification steps
1. Checkout this branch
2. Install RHOAM to remote cluster and wait for the installation to complete:
```
export INSTALLATION_TYPE=managed-api
make cluster/prepare/local 
make code/run
```
3. Verify that the following six probes are present in the `redhat-rhoam-observability` namespace:
![probe-instances](https://user-images.githubusercontent.com/80267718/134976144-7d5204da-bf89-47fb-9957-7a895eab59b8.png)
4. Verify that the probes can be seen on the Targets page in the `redhat-rhoam-observability` Prometheus instance and that when expanded, each target lists its State as UP:
![prometheus-probe-targets](https://user-images.githubusercontent.com/80267718/134977013-a39f6b45-748f-411f-93cc-842a21f3ab48.png)
